### PR TITLE
Parallel tests in CI

### PR DESCRIPTION
**Pull request by crdanielbusch**
## Description

Work in progress.

## Checklist

Please confirm that this pull request has done the following:

- [ ] Tests added
- [ ] Documentation added (where applicable)

### CODE REVIEW

**Comment by crdanielbusch** [comment]
@lewisjared can you have a look at this? 

I am not sure of all the steps are needed. I feel like there is not really a time improvement from running the tests in parallel.

Also there is an issue which I was not able to solve in the last job.

**Review by lewisjared — COMMENTED** [review]
I've made some suggested changes in #26  as it was easier to see if the suggestions actually work 😁

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -35,18 +44,113 @@ jobs:
...
+            type=schedule
+            type=ref,event=branch
+            type=ref,event=pr
+      - name: Build and push image
+        uses: docker/build-push-action@v5
+        id: build
+        with:
+          build-args: |
+            WRF_VERSION=${{ env.WRF_VERSION }}
+            WPS_VERSION=${{ env.WPS_VERSION }}
```
[the remark this answers is not in the document]
Comment (lewisjared) [reply .github/workflows/build.yaml:62]: The build isn't dependent on these variables

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -70,49 +174,79 @@ jobs:
...
         if: ${{ always() }}
         run: |
-          tar -cvzf results.tar.gz --exclude data/geog data
+          tar -cvzf results.tar.gz
       - name: Upload WRF artifacts
         if: ${{ always() }}
         uses: actions/upload-artifact@v4
         with:
           name: results
           path: ${{ github.workspace }}/results.tar.gz
```
[the remark this answers is not in the document]
Comment (lewisjared) [reply .github/workflows/build.yaml:249]: Fixed in #26. The arguments are important

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -29,11 +27,13 @@ jobs:
       contents: read
       packages: write
 
-    # Builds, tests and pushes the container
-    # This is performed via a single GH job because sharing images between jobs is somewhat clumsy.
+    outputs:
+      imageid: ${{ steps.build.outputs.imageid }}
```
Comment (lewisjared) [inline .github/workflows/build.yaml:31]: This line isn't needed anymore

**Review by lewisjared — APPROVED** [review]

**Pull request merged**